File: Data Structure & Algorithms/Patterns/Pattern 16/Problem16_Solution.js

function printPattern(num){
    for(let i=1;i<=num;++i){
        let start=1,spaces=(2*num)-(2*i)-1;
        for(let j=1;j<=i;++j){
            process.stdout.write(start+"\t");
            if(j!=i)
            start++;
        }
        
        for(let j=1;j<=spaces;++j){
            process.stdout.write("\t");
        }
        
        if(start==num){
            start--;
        }
        
        for(let j=1;j<=i;++j){
            if(j!=num)
            {
                process.stdout.write(start+"\t");
                start--;
            }
        }
        
        console.log();
    }
}

printPattern(7);

File: DSA_Level-1/Basics Of Programming/Patterns/Pattern 16/Problem16_Solution.java
import java.util.*;

public class Main{

public static void main(String[] args) {
    Scanner scn = new Scanner(System.in);
    int num=scn.nextInt();
    for(int i=1;i<=num;++i){
        int start=1,spaces=(2*num)-(2*i)-1;
        for(int j=1;j<=i;++j){
            System.out.print(start+"	");
            if(j!=i)
            start++;
        }
        
        for(int j=1;j<=spaces;++j){
            System.out.print("	");
        }
        
        if(start==num){
            start--;
        }
        
        for(int j=1;j<=i;++j){
            if(j!=num)
            {
                System.out.print(start+"	");
                start--;
            }
        }
        
        System.out.println();
    }
 }
}